locksmith/__tests__/controllers/purchaseController.test.ts

```diff
@@ -0,0 +1,71 @@
+const request = require('supertest')
+const ethJsUtil = require('ethereumjs-util')
+const sigUtil = require('eth-sig-util')
+const app = require('../../src/app')
+const Base64 = require('../../src/utils/base64')
+
+let privateKey = ethJsUtil.toBuffer(
+  '0xfd8abdd241b9e7679e3ef88f05b31545816d6fbcaf11e86ebd5a57ba281ce229'
+)
+
+function generateTypedData(message: any) {
+  return {
+    types: {
+      EIP712Domain: [
+        { name: 'name', type: 'string' },
+        { name: 'version', type: 'string' },
+        { name: 'chainId', type: 'uint256' },
+        { name: 'verifyingContract', type: 'address' },
+        { name: 'salt', type: 'bytes32' },
+      ],
+      PurchaseRequest: [
+        { name: 'recipient', type: 'address' },
+        { name: 'lock', type: 'address' },
+        { name: 'expiry', type: 'uint64' },
+      ],
+    },
+    domain: {
+      name: 'Unlock',
+      version: '1',
+    },
+    primaryType: 'PurchaseRequest',
+    message: message,
+  }
+}
+
+describe('Purchase Controller', () => {
+  describe('purchase initiation', () => {
+    describe('when the purchase request is appropriately signed', () => {
+      it('returns a 401 status code', async () => {
```

Comment:
It reads like this returns `401 Unauthorized` when the key is signed the right way, and `202 Accepted` when the key is signed the wrong way. I feel like this must be a case of swapped descriptions?

Reply:
good catch, i've updated this to reflect the feedback